client/state/ui/reducer.js

```diff
@@ -29,6 +29,26 @@ export function selectedSiteId( state = null, action ) {
 	return state;
 }
 
+/**
+ * Tracks the four most recently selected site IDs.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export function setRecentlySelectedSiteId( state = [], action ) {
```

Comment:
I would expect the naming of this to mirror the storing of data, not the action, i.e. `recentlySelectedSiteIds`